### PR TITLE
Glossary: adding selectors and variables

### LINKED ISSUE #23616: Missing in Glossary
### Feedback

I find these things missing in the Glossary:
- Selectors
- Variables

### URL

https://www.home-assistant.io/docs/glossary/

### Version

2022.7.7

### Additional information

Thanks for the hard work!

### PR DESCRIPTION
**Pull request by pyrrolizin**
## Proposed change

Adding Selectors and Variables to the glossary.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #23616

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by jbouwh — APPROVED** [review]
Nice work @pyrrolizin!

File: source/_data/glossary.yml [hunk 1]
```diff
@@ -60,6 +60,8 @@
   description: "[Scripts](/docs/scripts/) are components that allow users to specify a sequence of actions to be executed by Home Assistant when turned on."
 - topic: Sensor
   description: "[Sensors](/integrations/sensor) return information about a thing, for instance the level of water in a tank."
+- topic: Selectors
+  description: "[Selectors](/docs/blueprint/selectors/) specify what values are accepted for a blueprint input and how the input is shown in the user interface."
```
Comment (frenck) [inline source/_data/glossary.yml:64]:
```suggestion
  description: "[Selectors](/docs/blueprint/selectors/) specify what values are accepted for a Blueprint input and how the input is shown in the user interface."
```

Not just Blueprints, they also apply to scripts, automations, and services.
They don't specify values, but they provide UI features/input.

File: source/_data/glossary.yml [hunk 2]
```diff
@@ -70,5 +72,7 @@
   description: "A [trigger](/docs/automation/trigger/) is a set of values or conditions of a platform that are defined to cause an automation to run."
 - topic: TTS
   description: "TTS ([text to speech](/integrations/tts)) allows Home Assistant to talk to you."
+- topic: Variables
+  description: "[Variables](/docs/scripts/#variables) are a named storage for a value that can be processed e.g. in a script."
```
Comment (frenck) [inline source/_data/glossary.yml:76]: I think "storage" is a wrong term, as they are not stored at all.

**Review by frenck — CHANGES_REQUESTED** [review]
Please address these comments in a new PR @pyrrolizin

**Comment by pyrrolizin** [comment]
> Please address these comments in a new PR @pyrrolizin

I will take care of it. :+1:

File: source/_data/glossary.yml (hunk 1 above)
Comment (pyrrolizin) [reply source/_data/glossary.yml:64]: Changed. I took one of the sentence from the official documentation.
https://www.home-assistant.io/docs/blueprint/selectors/
See #23737

File: source/_data/glossary.yml (hunk 2 above)
Comment (pyrrolizin) [reply source/_data/glossary.yml:76]: Changed. You are right. it is not classic persistent storage and this could be misunderstood.
See #23737

**Pull request merged**